sycl/include/syclcompat/launch_experimental.hpp

```diff
@@ -0,0 +1,227 @@
+ *    Depends on the following 3 extensions-
+ *       sycl_ext_oneapi_kernel_properties
+ *       sycl_ext_oneapi_enqueue_functions
+ *       sycl_ext_oneapi_properties
+ **************************************************************************/
+
+#pragma once
+
+#include <sycl/ext/oneapi/experimental/enqueue_functions.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+
+#include <sycl/sycl.hpp>
+
+#include <tuple>
+
+#include <syclcompat/device.hpp>
+#include <syclcompat/dims.hpp>
+#include <syclcompat/kernel_properties.hpp>
+#include <syclcompat/launch.hpp>
+
+#if defined(SYCL_EXT_ONEAPI_KERNEL_PROPERTIES) &&                              \
+    defined(SYCL_EXT_ONEAPI_PROPERTIES)
+// defined(SYCL_EXT_ONEAPI_ENQUEUE_FUNCTIONS) uncomment once
+// SYCL_EXT_ONEAPI_ENQUEUE_FUNCTIONS is defined
+
+namespace sycl_exp = sycl::ext::oneapi::experimental;
+
+namespace syclcompat {
+namespace experimental {
+
+namespace detail {
+constexpr auto empty_property_list =
+    sycl::ext::oneapi::experimental::properties{};
+}
+
+//================================================================================================//
+                      // Overloads using Local Memory //
+//================================================================================================//
+
+template <auto KernelFunctor, typename PropertyList, typename... Args>
+inline std::enable_if_t<std::is_invocable_v<decltype(KernelFunctor), Args..., char *>,
+                        sycl::event>
+launch(const sycl::nd_range<3> &launch_params, std::size_t local_memory_size,
+       const PropertyList &launch_properties, const sycl::queue &queue,
+       const Args &...args) {
+  sycl_exp::launch_config config(launch_params, launch_properties);
+  return sycl_exp::submit_with_event([&](sycl::handler &cgh) {
+    sycl::local_accessor<char, 1> local_memory(local_memory_size, cgh);
+    sycl_exp::nd_launch(cgh, config,
+                        KernelFunctor(arg..., local_memory.get_pointer()));
+  });
+}
+
+template <int Dim, auto KernelFunctor, typename PropertyList, typename... Args>
+inline std::enable_if_t<std::is_invocable_v<decltype(KernelFunctor), Args..., char *>,
+                        sycl::event>
+launch(const sycl::nd_range<Dim> &launch_params, std::size_t local_memory_size,
+       const PropertyList &launch_properties, const Args &...args) {
+  return launch<KernelFunctor>(
+      ::syclcompat::detail::transform_nd_range(launch_params),
+      local_memory_size, launch_properties, ::syclcompat::get_default_queue(),
+      args...);
+}
+
+template <int Dim, auto KernelFunctor, typename... Args>
+inline std::enable_if_t<std::is_invocable_v<decltype(KernelFunctor), Args..., char *>,
+                        sycl::event>
+launch(const sycl::nd_range<Dim> &launch_params, std::size_t local_memory_size,
+       const Args &...args) {
+  using PropertyList = decltype(detail::empty_property_list);
+  return launch<KernelFunctor, PropertyList>(
```

Comment:
The overloads you have provided which don't take a `PropertyList` and instead pass an empty one: these are nice because they reflect how the `syclcompat::launch` functions will work once we integrate this properly. However, for now they are just duplicating the equivalent `syclcompat::launch` API but *without tests*. If you don't want to bother adding more tests for equivalent APIs, I would suggest pulling out these overloads.